[exporter/dynatrace] document resource_to_telemetry_conversion option (#13371)
open-telemetry/opentelemetry-collector-contrib: exporter/dynatraceexporter/README.md

Default: `5000`

### resource_to_telemetry_conversion (Optional)

When `resource_to_telemetry_conversion.enabled` is set to `true`, all resource
attributes will be included as metric attributes.

> :warning: **Please note** that the Dynatrace API has a limit of `50` attributes
> per metric and any metric which exceeds this limit will be dropped.
> If you think you might exceed this limit, you should use the

> > [attributes processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/attributesprocessor)

Reviewer: Does the attributes processor actually allow "promoting" resource attributes to metric attributes?  IIRC, it just allowed modifying metric attributes.

Reviewer 2: Ya I think you'd need to use the transform processor if you want to move resource attributes to datapoint attributes and then limit the number of attributes.

Author: Ah I must have been reading too quickly. Looking at the transform processor I can't see an obvious way to promote resource keys to attributes, but I am not really a TQL expert. Can you point me in the right direction?

Reviewer 2: It would look something like:

```
set(attributes["attribute key 1"], resource.attributes["attribute key 1"])
.
.
.
.
set(attributes["attribute key N"], resource.attributes["attribute key N"])
limit(attributes, 50)
```

There isn't a function that can copy them all over at once, but seems like that could be a supported function.

Author: I decided to go with `keep_keys` and enable `resource_to_telemetry_conversion`

Reviewer 2: I think _technically_ you could still end up with more than 50 attributes on data points with that technique.  If the data point has 48 attributes and you use `keep_keys` on `resource.attributes` to reduce the number of resource attributes to 3, and then use `resource_to_telemetry_conversion`, the data point will end up with 51 attributes.

Author: the problem with using `set` and `limit` is that dropping keys automatically is essentially impossible to do in an intelligent way. I think if users find themselves needing that level of customization they've progressed beyond something we're going to suggest in a README file and should be more comfortable with advanced collector configuration already.

Default: `5000`

### resource_to_telemetry_conversion (Optional)

When `resource_to_telemetry_conversion.enabled` is set to `true`, all resource

> attributes will be included as metric attributes.

Reviewer 3: ```suggestion
attributes will be included as metric dimensions in Dynatrace.
Note that these are added _in addition_ to your existing data point attributes.
```

> attributes will be included as metric attributes.

Reviewer 3: Can you specify which of the two takes precedence in case the attribute names would clash?

Reviewer 3: I noticed that this is not documented in the `resourcetotelemetry` helper's readme, so we should actually not mention the current precedence here, as it would be unedfined behavior that can change at any time.

Author: yeah i'll remove it

Default: `false`

> :warning: **Please note** that the Dynatrace API has a limit of `50` attributes

> > per metric and any metric which exceeds this limit will be dropped.

Reviewer 3: ```suggestion
> per metric data point and any data point which exceeds this limit will be dropped.
```

If you think you might exceed this limit, you should use the
[transform processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/transformprocessor)

>  to remove any resource attributes you don't want converted.

Reviewer 3: ```suggestion
to remove any resource attributes you don't want to be converted.
```

>  to remove any resource attributes you don't want converted.

Reviewer 3: In your example, you don't specify the ones you drop, but rather supply a filter for which ones you keep, so maybe:
```suggestion
to apply a filter, so only a select subset of your resource attributes are converted.
```
